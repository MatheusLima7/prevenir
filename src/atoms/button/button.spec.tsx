import { render } from "@testing-library/react-native";

import Button from "./button";

jest.mock("react-native-linear-gradient", () => {
  const { View } = require("react-native");
  return View;
});

describe("Button Component", () => {
  it("should render component correctly", () => {
    const container = render(
      <Button mode="full" onPress={() => jest.fn()} type="filled" size="small">
        teste
      </Button>
    );

    expect(container).toMatchSnapshot();
  });

  it("should to render the button with attributes of filled button", () => {
    const { getByText } = render(
      <Button mode="full" onPress={() => jest.fn()} type="filled" size="small">
        teste
      </Button>
    );
    const textComponent = getByText("teste");
    const style = textComponent.props.style;
    const mergedStyle = Object.assign({}, ...style);

    expect(mergedStyle).toHaveProperty("fontFamily", "Quicksand-SemiBold");
    expect(mergedStyle).toHaveProperty("fontSize", 14);
    expect(mergedStyle).toHaveProperty("fontWeight", `600`);
  });
});
