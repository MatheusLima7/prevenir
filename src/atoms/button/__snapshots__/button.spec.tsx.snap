// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should render component correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "alignItens": "center",
      "backgroundColor": "#007AFF",
      "borderBottomLeftRadius": 7,
      "borderBottomRightRadius": 7,
      "borderTopLeftRadius": 7,
      "borderTopRightRadius": 7,
      "color": "#ffffff",
      "display": "flex",
      "flexDirection": "row",
      "fontSize": 14,
      "height": 60,
      "justifyContent": "space-evenly",
      "lineHeight": 20,
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 30,
      "paddingRight": 30,
      "paddingTop": 0,
      "width": "100%",
    }
  }
>
  <Text
    center={false}
    color="white"
    size="small"
    style={
      [
        {
          "color": "#ffffff",
          "display": "flex",
          "fontFamily": "Quicksand-SemiBold",
          "fontSize": 14,
          "fontWeight": "600",
          "lineHeight": 20,
        },
        {
          "width": "100%",
        },
      ]
    }
    testId="text-component"
    type="title"
  >
    teste
  </Text>
</View>
`;
